CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+
+project(pure-maps
+	VERSION 2.6.5
+	DESCRIPTION "Maps and navigation")
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+include(FeatureSummary)
+include(GNUInstallDirs)
+include(CMakePrintHelpers)
+
+option(USE_BUNDLED_GPXPY "Use a bundled version of GPXPY rather than a system-wide version" ON)
+option(FLAVOR "Platform to build support for. Supported platforms: kirigami, silica, qtcontrols, uuitk" OFF)
+set(PROFILE "Online" CACHE STRING "Default profile. Supported profiles: Online, Mixed, Offline")
+option(RUN_FROM_SOURCE "Run from source, this is mainly intended for easy development" OFF)
+
+set(VALID_FLAVOR_OPTIONS
+	"kirigami"
+	"silica"
+	"qtcontrols"
+	"uuitk")
+
+if(NOT FLAVOR IN_LIST VALID_FLAVOR_OPTIONS)
+	message(ERROR " Please specify a valid platform using -DFLAVOR=platform as CMake option!")
+	return()
+endif()
+
+if(PROFILE STREQUAL "Online" OR PROFILE STREQUAL "Mixed")
+	set(DEFAULT_BASEMAP "Mapbox" CACHE STRING "Default basemap")
+	set(DEFAULT_GEOCODER "photon" CACHE STRING "Default geocoder")
+	set(DEFAULT_GUIDE "foursquare" CACHE STRING "Default guide")
+	set(DEFAULT_ROUTER "stadiamaps" CACHE STRING "Default router")
+elseif(PROFILE STREQUAL "Offline")
+	set(DEFAULT_BASEMAP "OSM Scout" CACHE STRING "Default basemap")
+	set(DEFAULT_GEOCODER "osmscout" CACHE STRING "Default geocoder")
+	set(DEFAULT_GUIDE "osmscout" CACHE STRING "Default guide")
+	set(DEFAULT_ROUTER "osmscout" CACHE STRING "Default router")
+else()
+	message(ERROR " Please specify a supported profile using -DPROFILE as CMake option!")
+	return()
+endif()
+
+if(FLAVOR STREQUAL "silica")
+	set(APP_NAME harbour-pure-maps)
+else()
+	set(APP_NAME pure-maps)
+endif()
+
+set(QT_MIN_VERSION "5.6.0")
+find_package(Python3 COMPONENTS Interpreter REQUIRED)
+find_package(Gettext REQUIRED)
+find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS Gui Positioning DBus LinguistTools REQUIRED)
+
+if(FLAVOR STREQUAL "kirigami" OR FLAVOR STREQUAL "qtcontrols" OR FLAVOR STREQUAL "uuitk")
+	find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS Quick Qml Widgets QuickControls2 REQUIRED)
+endif()
+
+if(RUN_FROM_SOURCE)
+	set(DATADIR ${CMAKE_CURRENT_SOURCE_DIR})
+	set(DATADIR_RUNNING ${DATADIR})
+
+	add_custom_command(OUTPUT .platformlink.set.${FLAVOR}
+		COMMAND rm -f ${CMAKE_CURRENT_SOURCE_DIR}/qml/platform .platformlink.set.*
+		COMMAND ln -s platform.${FLAVOR} ${CMAKE_CURRENT_SOURCE_DIR}/qml/platform
+		COMMAND touch .platformlink.set.${FLAVOR})
+	add_custom_target(run_from_source ALL
+		DEPENDS .platformlink.set.${FLAVOR})
+	message(WARNING "Please DO NOT run 'cmake --install' in this build")
+else()
+	set(DATADIR ${CMAKE_INSTALL_FULL_DATAROOTDIR}/${APP_NAME})	
+	if(FLAVOR STREQUAL "uuitk")
+		set(DATADIR_RUNNING ".")
+	else()
+		set(DATADIR_RUNNING ${DATADIR})
+	endif()
+endif()
+
+add_subdirectory(src)
+add_subdirectory(poor)
+add_subdirectory(thirdparty)
+add_subdirectory(qml)
+add_subdirectory(po)
+
+install(DIRECTORY geocoders
+	DESTINATION ${DATADIR})
```

Comment:
This installs goodies like `__pycache__` and tests which are not really needed on device. Corresponding selection in PRO file was something like `geocoders/*.py geocoders/*.json geocoders/README.md`. If you could demo how to make it properly, I can fix them all as well.

See https://github.com/rinigus/pure-maps/blob/master/pure-maps.pro#L127 and below